chat-plugins/sapience-plugins.js

```diff
@@ -0,0 +1,199 @@
+			method: 'GET',
+		};
+		let req = http.request(reqOpt, function(res) {
+			res.on('data', function(chunk) {
+				try {
+					let data = JSON.parse(chunk);
+					let output = '<center><img src="' + data.data["image_url"] + '" width="75%"></center>';
+					if (!self.runBroadcast()) return;
+					if (data.data["image_url"] === undefined) {
+						self.errorReply("ERROR CODE 404: No images found!");
+						return room.update();
+					} else {
+						self.sendReplyBox(output);
+						return room.update();
+					}
+				} catch (e) {
+					self.errorReply("ERROR CODE 503: Giphy is unavaliable right now. Try again later."); //Anti Spam Filter
+					return room.update();
+				}
+			});
+		});
+		req.end();
+	},
+	animalshelp: ['|html|<div class="infobox">Animals Plugin by <font color="orange">Sapience</font>' +
+		'<ul><li>/animals cat - Displays a cat.</li>' +
+		'<li>/animals kitten - Displays a kitten.</li>' +
+		'<li>/animals dog - Displays a dog.</li>' +
+		'<li>/animals puppy - Displays a puppy.</li>' +
+		'<li>/animals bunny - Displays a bunny.</li>' +
+		'<li>/animals otter - Displays an otter.</li>' +
+		'<li>/animals pokemon - Displays a pokemon.</li>' +
+		'<li>/animals help - Displays this help box.</li></ul></div>'
+],
+	gif: 'giphy',
+	giphy: function(target, room, user) {
+		if (!target) return this.parse('/help giphy')
+		let tarId = toId(target);
+		let self = this;
+		let reqOpt = {
+			hostname: 'api.giphy.com', // Do not change this
+			path: '/v1/gifs/random?api_key=dc6zaTOxFJmzC&tag=' + tarId,
+			method: 'GET',
+		};
+		let req = http.request(reqOpt, function(res) {
+			res.on('data', function(chunk) {
+				try {
+					let data = JSON.parse(chunk);
+					let output = '<center><img src="' + data.data["image_url"] + '" width="50%"></center>';
+					if (!self.runBroadcast()) return;
+					if (data.data["image_url"] === undefined) {
+						self.errorReply("ERROR CODE 404: No images found!");
+						return room.update();
+					} else {
+						self.sendReplyBox(output);
+						return room.update();
+					}
+				} catch (e) {
+					self.errorReply("ERROR CODE 503: Giphy is unavaliable right now. Try again later.");
+					return room.update();
+				}
+			});
+		});
+		req.end();
+	},
+	giphyhelp: ['- /giphy [word] - Displays a random Gif with your chosen keyword.'],
+	
+	
+	
+/****************************************
+* Events Plugin by RoboPhill & Sapience *
+****************************************/
+
+    event: {
+        create: function (target, room, user) {
+            let params = target.split(", ") || target.split(",");
+            if (!target || !params[0] || !params[1] || !params[2]) return this.errorReply('Usage: /event create [Name], [Desc], [Room]');
+            if (params[1].length > 500) return this.errorReply('Descriptions must be a maximum of 500 characters!');
+            if (EventsOn[toId(params[0])]) return this.sendReply("/event - The event \"" + params[0] + "\" already exists.");
```

Comment:
Even though you added this check, you didn't set the value of the <code>toId(params[0])</code> key anywhere else in the script. You only set values for <code>Name</code>, <code>DisplayName</code>, <code>Desc</code, and <code>Room</code>